src/main/kotlin/cacophony/grammars/Analysis.kt

```diff
@@ -2,37 +2,34 @@ package cacophony.grammars
 
 import cacophony.automata.DFA
 
-// TODO: Uncomment this after #46
-// typealias DFAStateReference<StateType, Atom, Result>
-//     = Pair<StateType, DFA<StateType, Atom, Result>>
-typealias DFAStateReference<StateType> = Pair<StateType, DFA<StateType>>
+typealias DFAStateReference<StateType, AtomType, ResultType> = Pair<StateType, DFA<StateType, AtomType, ResultType>>
 
-fun <StateType, SymbolType> findNullable(automata: Map<SymbolType, DFA<StateType>>): Collection<DFAStateReference<StateType>> {
+fun <StateType, SymbolType> findNullable(
+    automata: Map<SymbolType, DFA<StateType, *, SymbolType>>,
+): Collection<DFAStateReference<StateType, *, SymbolType>> {
     TODO("Not implemented")
 }
 
 fun <StateType, SymbolType> findFirst(
-    automata: Map<SymbolType, DFA<StateType>>,
-    nullable: Collection<DFAStateReference<StateType>>,
-): Map<DFAStateReference<StateType>, Collection<SymbolType>> {
+    automata: Map<SymbolType, DFA<StateType, *, SymbolType>>,
+    nullable: Collection<DFAStateReference<StateType, *, SymbolType>>,
+): Map<DFAStateReference<StateType, *, SymbolType>, Collection<SymbolType>> {
     TODO("Not implemented")
 }
 
 fun <StateType, SymbolType> findFollow(
-    automata: Map<SymbolType, DFA<StateType>>,
-    nullable: Collection<DFAStateReference<StateType>>,
-    first: Map<DFAStateReference<StateType>, Collection<SymbolType>>,
-): Map<DFAStateReference<StateType>, Collection<SymbolType>> {
+    automata: Map<SymbolType, DFA<StateType, *, SymbolType>>,
+    nullable: Collection<DFAStateReference<StateType, *, SymbolType>>,
+    first: Map<DFAStateReference<StateType, *, SymbolType>, Collection<SymbolType>>,
+): Map<DFAStateReference<StateType, *, SymbolType>, Collection<SymbolType>> {
     TODO("Not implemented")
 }
 
 data class AnalyzedGrammar<StateType, SymbolType>(
     val syncSymbolTypes: Collection<SymbolType>,
-    val automata: Map<SymbolType, DFA<StateType>>,
-    // TODO: Uncomment after #46
-    // val automata: Map<SymbolType, DFA<StateType, Atom, SymbolType>>
+    val automata: Map<SymbolType, DFA<StateType, *, SymbolType>>,
```

Comment:
this is not the valid type, it should be

```kotlin
val automata: Map<SymbolType, DFA<StateType, SymbolType, Production<SymbolType>>>
```

because these the alphabet of these automata are symbols in our grammars (`SymbolType`), and the result is from which production the match was found (so the result is just `Production<SymbolType>`)